inc/Ui/DiffView.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace dokuwiki\Ui;
+
+use dokuwiki\ChangeLog\RevisionInfo;
+use dokuwiki\Ui\PageDiff;
+use dokuwiki\Ui\MediaDiff;
+
+/**
+ * DokuWiki Generic Diff Viewer
+ *
+ * @package dokuwiki\Ui
+ */
+class DiffView extends Ui
+{
+    /* @var string */
+    protected $id;   // page id or media id
+
+    /* @var string */
+    protected $mode;   // page or media
+
+    /* @var DiffViewInterface */
+    protected $diff;
+
+    /**
+     * DiffView constructor
+     *
+     * @param string $id  page id or media id
+     */
+    public function __construct($id)
+    {
+      //$this->id = $id;
+        if (empty($id)) {
+            $this->mode = 'media';
+        } else {
+            $this->mode = strrpos($id, '.') ? 'media' : 'page';
+        }
+
+        // diff viewer switching strategy
+        $this->diff = $this->getDiffViewInterface($id);
+    }
+
+    /**
+     * Get instance of diff view interface
+     *
+     * @param string $id  page id or media id
+     */
+    protected function getDiffViewInterface($id)
+    {
+        switch ($this->mode) {
+            case 'page' : return new PageDiff($id);
+            case 'media': return new MediaDiff($id);
+        }
+    }
+
+
+    /**
+     * Set a pair of revisions to be compared
+     *
+     * @param int $rev1 older revision
+     * @param int $rev2 newer revision
+     * @return $this
+     */
+    public function compare($rev1, $rev2)
+    {
+        $diff =& $this->diff;
```

Comment:
Is it not more practical to use just `$this->diff` instead of this alias?
It looks now more as local variable, while it is actual class variable.

Reply:
I see. Thanks.